### DISCUSSION TITLE
Welcome to Getbhavcopy-alternative Discussions!

### DISCUSSION THREAD

**Discussion by pparesh25** [pparesh25/Getbhavcopy-alternative#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jayzarna** [comment]
Hi Paresh, downloaded the files and tried to download the NSE Bhavcopy files but it did not download anything.

It made the directory but in the directory nothing was found. 

I have a Windows 11 machine. Regards Jayesh

**Reply by svgangal** [reply]
Hi Paresh,

I am an American, but originally from Mumbai.

You have published very nice work.  Please accept my compliments.  

I'd like to donate, but I am unaware how to donate via UPI.

Please let me know.  My email is: salil.gangal@gmail.com

BTW I am a trader/coder and my site is:  http://salilgangal.com

Thanks.
~ Salil

**Reply by pparesh25** [reply]
I sincerely apologize for the late reply. It is an honor that you saw my work and offered such positive feedback. I visited your website, which is a vast source of knowledge for me, and I have much to learn from it.

Unfortunately, I don't currently have any option beyond UPI to accept donations. As you know, utilizing UPI requires an Indian bank account, which wouldn't be feasible for you as a US citizen. However, I truly appreciate your desire to support my work.

**Comment by jad-aditya** [comment]
Hi Paresh,
Great work you have done over here.
However, I'm facing an unusual error.
My Index data is getting converted into Metastock but not equity data

**Reply by pparesh25** [reply]
The code is working well and I am not getting any such feedback from other users.

**Comment by ghkumarchd** [comment]
Hi Paresh, when nse working on saturday or sunday  i can't download data  any restrictions you put on code on these days other working days data is download successfully

**Reply by pparesh25** [reply]
Added Weekend_Getbhavcopy-Eq-with original Indexes name.py file to download data in weekend.
But my recommendation is to use this to download data only on weekends when the market is open.

**Comment by Rvgandhi** [comment]
Not able to download the NSE Bhavcopy today ; Has NSE changed its file structure again ?

**Reply by pparesh25** [reply]
A new file has been added with name "New-Getbhavcopy-Eq-with original Indexes name.py " use it to download data after 8 july 2024.
(To implement a new format, NSE has discontinued the old link for data download starting from July 8, 2024, and introduced a new format on a new link. Consequently, necessary changes have been made in the code to accommodate this update. A new file named "New-Getbhavcopy-Eq-with original Indexes name.py" has been created to download data after July 8, 2024.

However, the "Getbhavcopy-Eq-with original Indexes name.py" file will continue to be used to download historical data, as data prior to July 8, 2024, remains available through the old link)

**Comment by DatapulseEOD** [comment]
For users who mainly want the final NSE EOD BhavCopy file rather than running scripts, I publish the processed data every day. Files are neat OHLC text (Open, High, Low, Close, volume, etc.), ready to plug into Amibroker and MetaStock without extra formatting. Daily NSE EOD BhavCopy is available for download .

https://t.me/+A_RO35EQ-7o4YmJl

Check this 
[2026-03-17-NSE-EQ.txt](https://github.com/user-attachments/files/26178473/2026-03-17-NSE-EQ.txt)
[2026-03-18-NSE-EQ.txt](https://github.com/user-attachments/files/26178475/2026-03-18-NSE-EQ.txt)
[2026-03-19-NSE-EQ.txt](https://github.com/user-attachments/files/26178476/2026-03-19-NSE-EQ.txt)
[2026-03-20-NSE-EQ.txt](https://github.com/user-attachments/files/26178477/2026-03-20-NSE-EQ.txt)
